presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/BingTileFunctions.java

```diff
@@ -371,16 +391,92 @@ private static Point tileXYToLatitudeLongitude(int tileX, int tileY, int zoomLev
         return new Point(longitude, latitude);
     }
 
+    /**
+     * Returns a Bing tile at a given zoom level containing a point at a given latitude and longitude.
+     * Latitude must be within [-85.05112878, 85.05112878] range. Longitude must be within [-180, 180] range.
+     * Zoom levels from 1 to 23 are supported.
+     */
     private static BingTile latitudeLongitudeToTile(double latitude, double longitude, int zoomLevel)
+    {
+        long mapSize = mapSize(zoomLevel);
+        int tileX = longitudeToTileX(longitude, mapSize);
+        int tileY = longitudeToTileY(latitude, mapSize);
+        return BingTile.fromCoordinates(tileX, tileY, zoomLevel);
+    }
+
+    /**
+     * Called by bing_tiles_around, return immediate neighbour surrounding a point at a given latitude and longitude.
+     */
+    private static Set<BingTile> latitudeLongitudeToSurroundingTiles(double latitude, double longitude, int zoomLevel)
+    {
+        return latitudeLongitudeToSurroundingTiles(latitude, longitude, zoomLevel, 1);
+    }
+
+    /**
+     * Returns a collection of tiles at a given zoom level surrounding a point at a given latitude and longitude.
+     * It first locate the tile in the center.
+     * then it goes around to collect the tiles nearby
+     */
+    private static Set<BingTile> latitudeLongitudeToSurroundingTiles(double latitude, double longitude, int zoomLevel, int radius)
+    {
+        long mapSize = mapSize(zoomLevel);
+        long maxTileIndex = (mapSize / TILE_PIXELS) - 1;
+
+        checkCondition(radius >= 0 && radius <= maxTileIndex, "radius should be between 0 and map size");
+
+        int tileX = longitudeToTileX(longitude, mapSize);
+        int tileY = longitudeToTileY(latitude, mapSize);
+
+        Set<BingTile> bingTiles = new HashSet<>();
+
+        for (int i = -1 * radius; i <= radius; i++) {
+            for (int j = -1 * radius; j <= radius; j++) {
+                if (coordinateInsideBoundary(tileX + i, maxTileIndex)
```

Comment:
Maybe we can just inline `coordinateInsideBoundary`? Also, I think this check can be extracted to the outer loop (loop for i).

i.e.

```java
for (int i = -radius; i <= radius; i++) {
    if (tileX + i < 0 || tileX + i > maxTileIndex) {
        continue;
    }
    for (int j = ...) {
    }
}
```

Reply:
??
Can you clarify